File: content-api/orchestrator/src/main/java/org/sunbird/actors/HealthActor.java
package org.sunbird.actors;

import org.sunbird.actor.core.BaseActor;
import org.sunbird.common.dto.Request;
import org.sunbird.common.dto.Response;
import org.sunbird.graph.health.HealthCheckManager;

import scala.concurrent.Future;

public class HealthActor extends BaseActor {

    @Override
    public Future<Response> onReceive(Request request) {
        return HealthCheckManager.checkAllSystemHealth(getContext().dispatcher());
    }

}

File: taxonomy-api/taxonomy-actors/src/main/scala/org/sunbird/actors/HealthActor.scala
package org.sunbird.actors

import javax.inject.Inject

import scala.concurrent.{ExecutionContext, Future}
import org.sunbird.actor.core.BaseActor
import org.sunbird.common.dto.{Request, Response, ResponseHandler}
import org.sunbird.graph.OntologyEngineContext
import org.sunbird.graph.health.HealthCheckManager


class HealthActor @Inject() (implicit oec: OntologyEngineContext) extends BaseActor {

    implicit val ec: ExecutionContext = getContext().dispatcher

    @throws[Throwable]
    override def onReceive(request: Request): Future[Response] = {
        HealthCheckManager.checkAllSystemHealth()
    }
}
